test/pxenforcer.test.js

```diff
@@ -0,0 +1,58 @@
+const { describe, it } = require("mocha");
+const { expect } = require("chai");
+const sinon = require("sinon");
+
+const PerimeterXEnforcer = require("../lib/pxenforcer");
+
+describe("PerimeterXEnforcer", () => {
+    const BASIC_CONFIG = {
+        px_app_id: 'PX_APP_ID',
+        px_cookie_secret: 'PX_COOKIE_SECRET',
+        px_auth_token: 'PX_AUTH_TOKEN'
+    };
+
+    describe("newMiddleware", () => {
+        it("should return a function that accepts 3 parameters (req, res, next)", () => {
+            // arrange
+            const pxEnforcer = new PerimeterXEnforcer(BASIC_CONFIG);
+
+            // act
+            const middleware = pxEnforcer.newMiddleware();
+
+            // assert
+            expect(middleware).to.be.a('function').with.lengthOf(3);
+        });
+    });
+
+    describe("sendAdditionalS2SActivity", () => {
```

Comment:
is there a reason we don't test other activity types?
anyway looks good!

Reply:
The reason is because this is the only activity that has a public function in the API